state/application.go

```diff
@@ -1689,13 +1689,21 @@ func applicationRelations(st *State, name string) (relations []*Relation, err er
 }
 
 // ConfigSettings returns the raw user configuration for the application's charm.
-// Unset values are omitted.
 func (a *Application) ConfigSettings() (charm.Settings, error) {
 	settings, err := readSettings(a.st.db(), settingsC, a.settingsKey())
 	if err != nil {
 		return nil, err
 	}
-	return settings.Map(), nil
+	chrm, _, err := a.Charm()
```

Comment:
can we refactor this and the Unit.ConfigSettings method to use a common function? it's all the same except the charm URL used to determine the settings key and fetch the charm

Reply:
done